log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationFactory.java

```diff
@@ -118,6 +121,15 @@ protected boolean isActive() {
         return true;
     }
 
+    /**
+     * Required for Spring Boot.
+     * @param props PropertiesUtil.
+     * @return the AuthorizationProvider, if any.
+     */
+    public static AuthorizationProvider authorizationProvider(final PropertiesUtil props) {
+        return AuthorizationProvider.getAuthorizationProvider((PropertyEnvironment) props);
+    }
```

Comment:
I agree that the method should not be duplicated.

Also, in order to make this PR back-portable to `2.x` (it is a requirement, isn't it?), I would use `LoggerContext` as parameter.

Unless I am mistaken, all HTTP/HTTPS accesses are performed **after** a `LoggerContext` is created and the context is passed to the `ConfigurationFactory` as a parameter (since 2.7). This change would allow 3.x to configure separate credentials for separate web applications.

Reply:
The authorization provider is always created based on properties, partly because it can be used to authenticate with a remote source that provides your logging config.

Reply:
@jvz,

Yes, but it could use a different `PropertyEnvironment` for each `LoggerContext`.

Anyway the reason @rgoers "introduced" this method in 3.x is that it was already present in 2.x, but disappeared from 3.x or was never ported to `main`.

Reply:
@ppkarwasz Correct. It is NOT a new method. Matt removed it a few weeks before I made this fix.